File: XPath/org/eclipse/wst/xml/xpath2/processor/internal/ast/XPathVisitor.cs
/// <summary>
///*****************************************************************************
/// Copyright (c) 2005, 2013 Andrea Bittau, University College London, and others
/// All rights reserved. This program and the accompanying materials
/// are made available under the terms of the Eclipse Public License 2.0
/// which accompanies this distribution, and is available at
/// https://www.eclipse.org/legal/epl-2.0/
/// 
/// SPDX-License-Identifier: EPL-2.0
/// 
/// Contributors:
///     Andrea Bittau - initial API and implementation from the PsychoPath XPath 2.0 
/// ******************************************************************************
/// </summary>

using xpath.org.eclipse.wst.xml.xpath2.processor.@internal.ast;

namespace org.eclipse.wst.xml.xpath2.processor.@internal.ast
{

	using XPath = org.eclipse.wst.xml.xpath2.processor.ast.XPath;

	/// <summary>
	/// Visitor class for XPath expressions.
	/// </summary>
	public interface XPathVisitor
	{
		/// <summary>
		/// Visit XPath.
		/// </summary>
		object visit(XPath xp);

		/// <summary>
		/// Visit ForExpr.
		/// </summary>
		object visit(ForExpr fex);

		/// <summary>
		/// Visit QuantifiedExpr.
		/// </summary>
		object visit(QuantifiedExpr qex);

		/// <summary>
		/// Visit IfExpr.
		/// </summary>
		object visit(IfExpr ifex);

		/// <summary>
		/// Visit OrExpr.
		/// </summary>
		object visit(OrExpr orex);

		/// <summary>
		/// Visit AndExpr.
		/// </summary>
		object visit(AndExpr andex);

		/// <summary>
		/// Visit CmpExpr.
		/// </summary>
		object visit(CmpExpr cmpex);

		/// <summary>
		/// Visit RangeExpr.
		/// </summary>
		object visit(RangeExpr rex);

		/// <summary>
		/// Visit AddExpr.
		/// </summary>
		object visit(AddExpr addex);

		/// <summary>
		/// Visit SubExpr.
		/// </summary>
		object visit(SubExpr subex);

		/// <summary>
		/// Visit MulExpr.
		/// </summary>
		object visit(MulExpr mulex);

		/// <summary>
		/// Visit DivExpr.
		/// </summary>
		object visit(DivExpr mulex);

		/// <summary>
		/// Visit IDivExpr.
		/// </summary>
		object visit(IDivExpr mulex);

		/// <summary>
		/// Visit ModExpr.
		/// </summary>
		object visit(ModExpr mulex);

		/// <summary>
		/// Visit UnionExpr.
		/// </summary>
		object visit(UnionExpr unex);

		/// <summary>
		/// Visit PipeExpr.
		/// </summary>
		object visit(PipeExpr pipex);

		/// <summary>
		/// Visit IntersectExpr.
		/// </summary>
		object visit(IntersectExpr iexpr);

		/// <summary>
		/// Visit ExceptExpr.
		/// </summary>
		object visit(ExceptExpr eexpr);

		/// <summary>
		/// Visit InstOfExpr.
		/// </summary>
		object visit(InstOfExpr ioexp);

		/// <summary>
		/// Visit TreatAsExpr.
		/// </summary>
		object visit(TreatAsExpr taexp);

		/// <summary>
		/// Visit CastableExpr.
		/// </summary>
		object visit(CastableExpr cexp);

		/// <summary>
		/// Visit CastExpr.
		/// </summary>
		object visit(CastExpr cexp);

		/// <summary>
		/// Visit MinusExpr.
		/// </summary>
		object visit(MinusExpr e);

		/// <summary>
		/// Visit PlusExpr.
		/// </summary>
		object visit(PlusExpr e);

		/// <summary>
		/// Visit XPathExpr.
		/// </summary>
		object visit(XPathExpr e);

		/// <summary>
		/// Visit ForwardStep.
		/// </summary>
		object visit(ForwardStep e);

		/// <summary>
		/// Visit ReverseStep.
		/// </summary>
		object visit(ReverseStep e);

		/// <summary>
		/// Visit NameTest.
		/// </summary>
		object visit(NameTest e);

		/// <summary>
		/// Visit VarRef.
		/// </summary>
		object visit(VarRef e);

		/// <summary>
		/// Visit StringLiteral.
		/// </summary>
		object visit(StringLiteral e);

		/// <summary>
		/// Visit IntegerLiteral.
		/// </summary>
		object visit(IntegerLiteral e);

		/// <summary>
		/// Visit DoubleLiteral.
		/// </summary>
		object visit(DoubleLiteral e);

		/// <summary>
		/// Visit DecimalLiteral.
		/// </summary>
		object visit(DecimalLiteral e);

		/// <summary>
		/// Visit ParExpr.
		/// </summary>
		object visit(ParExpr e);

		/// <summary>
		/// Visit CntxItemExpr.
		/// </summary>
		object visit(CntxItemExpr e);

		/// <summary>
		/// Visit FunctionCall.
		/// </summary>
		object visit(FunctionCall e);

		/// <summary>
		/// Visit SingleType.
		/// </summary>
		object visit(SingleType e);

		/// <summary>
		/// Visit SequenceType.
		/// </summary>
		object visit(SequenceType e);

		/// <summary>
		/// Visit ItemType.
		/// </summary>
		object visit(ItemType e);

		/// <summary>
		/// Visit AnyKindTest.
		/// </summary>
		object visit(AnyKindTest e);

		/// <summary>
		/// Visit DocumentTest.
		/// </summary>
		object visit(DocumentTest e);

		/// <summary>
		/// Visit TextTest.
		/// </summary>
		object visit(TextTest e);

		/// <summary>
		/// Visit CommentTest.
		/// </summary>
		object visit(CommentTest e);

		/// <summary>
		/// Visit PITest.
		/// </summary>
		object visit(PITest e);

		/// <summary>
		/// Visit AttributeTest.
		/// </summary>
		object visit(AttributeTest e);

		/// <summary>
		/// Visit SchemaAttrTest.
		/// </summary>
		object visit(SchemaAttrTest e);

		/// <summary>
		/// Visit ElementTest.
		/// </summary>
		object visit(ElementTest e);

		/// <summary>
		/// Visit SchemElemTest.
		/// </summary>
		object visit(SchemaElemTest e);

		/// <summary>
		/// Visit AxisStep.
		/// </summary>
		object visit(AxisStep e);

		/// <summary>
		/// Visit FilterExpr.
		/// </summary>
		object visit(FilterExpr e);


        object visit(PostfixExpr fex);

	}

}

File: AntlrTreeEditing/src-java/org/eclipse/wst/xml/xpath2/processor/internal/ast/XPathVisitor.java
/*******************************************************************************
 * Copyright (c) 2005, 2013 Andrea Bittau, University College London, and others
 * All rights reserved. This program and the accompanying materials
 * are made available under the terms of the Eclipse Public License 2.0
 * which accompanies this distribution, and is available at
 * https://www.eclipse.org/legal/epl-2.0/
 *
 * SPDX-License-Identifier: EPL-2.0
 *
 * Contributors:
 *     Andrea Bittau - initial API and implementation from the PsychoPath XPath 2.0 
 *******************************************************************************/

package org.eclipse.wst.xml.xpath2.processor.internal.ast;

import org.eclipse.wst.xml.xpath2.processor.ast.XPath;

/**
 * Visitor class for XPath expressions.
 */
public interface XPathVisitor {
	/**
	 * Visit XPath.
	 */
	public Object visit(XPath xp);

	/**
	 * Visit ForExpr.
	 */
	public Object visit(ForExpr fex);

	/**
	 * Visit QuantifiedExpr.
	 */
	public Object visit(QuantifiedExpr qex);

	/**
	 * Visit IfExpr.
	 */
	public Object visit(IfExpr ifex);

	/**
	 * Visit OrExpr.
	 */
	public Object visit(OrExpr orex);

	/**
	 * Visit AndExpr.
	 */
	public Object visit(AndExpr andex);

	/**
	 * Visit CmpExpr.
	 */
	public Object visit(CmpExpr cmpex);

	/**
	 * Visit RangeExpr.
	 */
	public Object visit(RangeExpr rex);

	/**
	 * Visit AddExpr.
	 */
	public Object visit(AddExpr addex);

	/**
	 * Visit SubExpr.
	 */
	public Object visit(SubExpr subex);

	/**
	 * Visit MulExpr.
	 */
	public Object visit(MulExpr mulex);

	/**
	 * Visit DivExpr.
	 */
	public Object visit(DivExpr mulex);

	/**
	 * Visit IDivExpr.
	 */
	public Object visit(IDivExpr mulex);

	/**
	 * Visit ModExpr.
	 */
	public Object visit(ModExpr mulex);

	/**
	 * Visit UnionExpr.
	 */
	public Object visit(UnionExpr unex);

	/**
	 * Visit PipeExpr.
	 */
	public Object visit(PipeExpr pipex);

	/**
	 * Visit IntersectExpr.
	 */
	public Object visit(IntersectExpr iexpr);

	/**
	 * Visit ExceptExpr.
	 */
	public Object visit(ExceptExpr eexpr);

	/**
	 * Visit InstOfExpr.
	 */
	public Object visit(InstOfExpr ioexp);

	/**
	 * Visit TreatAsExpr.
	 */
	public Object visit(TreatAsExpr taexp);

	/**
	 * Visit CastableExpr.
	 */
	public Object visit(CastableExpr cexp);

	/**
	 * Visit CastExpr.
	 */
	public Object visit(CastExpr cexp);

	/**
	 * Visit MinusExpr.
	 */
	public Object visit(MinusExpr e);

	/**
	 * Visit PlusExpr.
	 */
	public Object visit(PlusExpr e);

	/**
	 * Visit XPathExpr.
	 */
	public Object visit(XPathExpr e);

	/**
	 * Visit ForwardStep.
	 */
	public Object visit(ForwardStep e);

	/**
	 * Visit ReverseStep.
	 */
	public Object visit(ReverseStep e);

	/**
	 * Visit NameTest.
	 */
	public Object visit(NameTest e);

	/**
	 * Visit VarRef.
	 */
	public Object visit(VarRef e);

	/**
	 * Visit StringLiteral.
	 */
	public Object visit(StringLiteral e);

	/**
	 * Visit IntegerLiteral.
	 */
	public Object visit(IntegerLiteral e);

	/**
	 * Visit DoubleLiteral.
	 */
	public Object visit(DoubleLiteral e);

	/**
	 * Visit DecimalLiteral.
	 */
	public Object visit(DecimalLiteral e);

	/**
	 * Visit ParExpr.
	 */
	public Object visit(ParExpr e);

	/**
	 * Visit CntxItemExpr.
	 */
	public Object visit(CntxItemExpr e);

	/**
	 * Visit FunctionCall.
	 */
	public Object visit(FunctionCall e);

	/**
	 * Visit SingleType.
	 */
	public Object visit(SingleType e);

	/**
	 * Visit SequenceType.
	 */
	public Object visit(SequenceType e);

	/**
	 * Visit ItemType.
	 */
	public Object visit(ItemType e);

	/**
	 * Visit AnyKindTest.
	 */
	public Object visit(AnyKindTest e);

	/**
	 * Visit DocumentTest.
	 */
	public Object visit(DocumentTest e);

	/**
	 * Visit TextTest.
	 */
	public Object visit(TextTest e);

	/**
	 * Visit CommentTest.
	 */
	public Object visit(CommentTest e);

	/**
	 * Visit PITest.
	 */
	public Object visit(PITest e);

	/**
	 * Visit AttributeTest.
	 */
	public Object visit(AttributeTest e);

	/**
	 * Visit SchemaAttrTest.
	 */
	public Object visit(SchemaAttrTest e);

	/**
	 * Visit ElementTest.
	 */
	public Object visit(ElementTest e);

	/**
	 * Visit SchemElemTest.
	 */
	public Object visit(SchemaElemTest e);

	/**
	 * Visit AxisStep.
	 */
	public Object visit(AxisStep e);

	/**
	 * Visit FilterExpr.
	 */
	public Object visit(FilterExpr e);
}
